spring-boot/proxy/src/main/java/eu/coatrack/proxy/security/ApiKeyAuthTokenVerifier.java

```diff
@@ -140,6 +141,9 @@ private boolean isApiKeyValid(String apiKeyValue) throws AuthenticationException
                 if (Date.valueOf(LocalDate.now()).after(apiKey.getValidUntil())) {
                     throw new CredentialsExpiredException("Api key is expired");
                 }
+                if (apiKey.getDeletedWhen() != null) {
+                    throw new AccessDeniedException("access denied");
```

Comment:
Suggestion: return false here, then a BadCredentialsException will be thrown.

Reply:
I changed and tested.
It is working well.

Reply:
As soon as you accepted it, I will proceed to make the PR to the Main repository

Reply:
Looks good!

Reply:
OK!